src/ListingPage.php

```diff
@@ -136,6 +143,41 @@ private function set_filters(): void
         $this->context['current_post_type'] = $_GET['post-type'] ?? '';
     }
 
+    /**
+     * Add featured posts to the context for the News & Stories page.
+     */
+    private function set_featured_posts(): void
+    {
+        if (!is_home()) {
+            return;
+        }
+
+        $sticky_posts = get_option('sticky_posts');
+        if (empty($sticky_posts) || count($sticky_posts) < 4) {
+            return;
+        }
+
+        $featured_post_ids = array_slice($sticky_posts, 0, 4);
+        $this->context['featured_post_ids'] = $featured_post_ids;
+
+        $featured_query = new \WP_Query([
```

Comment:
This is a tricky implemetation since <core:query> doesn’t natively support post__in directly within the UI or most of the block settings.